app/controllers/check_ins_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CheckInsController < ApplicationController
+  # before_action :setup_variables, only: [:new, :create]
+  def index
+    @check_in = CheckInDecorator.new(check_in_params)
+  end
+
+  def create
+    if @check_in.save do |check_in_creation|
```

Comment:
Where is `@check_in` defined? In order to use a variable you have to first assign it; but there is no `@check_in = CheckIn.new(check_in_params)` in this method or any methods called before it.